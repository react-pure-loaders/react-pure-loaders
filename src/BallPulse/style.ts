import { keyframes } from '@emotion/react';
import { balls, globalAnimation } from '../mixins';
import delay from '../utilities';
import { Pulse } from '../types';

const scale = keyframes({
  '0%': {
    opacity: 1,
    transform: 'scale(1)',
  },
  '45%': {
    opacity: 0.7,
    transform: 'scale(0.1)',
  },
  '80%': {
    opacity: 1,
    transform: 'scale(1)',
  },
});

export function ballPulse() {
  const n = 3;
  let i = 1;
  const grid: Pulse = {};
  for (; i < 4; i += 1) {
    grid[`> div:nth-of-type(${i})`] = {
      animation: `${scale} 0.75s ${delay('0.12s', n, i)} infinite cubic-bezier(.2,.68,.18,1.08)`,
    };
  }

  return grid;
}

export default (color?: string) => ({
  ...ballPulse(),
  ' > div': {
    ...balls(color),
    ...globalAnimation(),
    display: 'inline-block',
  },
});
