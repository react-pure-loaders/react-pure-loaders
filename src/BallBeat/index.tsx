/** @jsx jsx */
import { jsx } from '@emotion/react';
import style from './style';
import { BaseProps } from '../types';

export default function BallBeat({ color, loading }: BaseProps) {
  if (loading) {
    return (
            <div css={style(color)}>
                <div/>
                <div/>
                <div/>
            </div>
    );
  }

  return null;
}
