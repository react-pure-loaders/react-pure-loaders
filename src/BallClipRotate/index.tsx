/** @jsx jsx */
import { jsx } from '@emotion/react';
import style from './style';
import { BaseProps } from '../types';

export default function BallClipRotate({ color, loading }: BaseProps) {
  if (loading) {
    return (
            <div css={style(color)} data-testid="ball-clip-rotate-container">
                <div/>
            </div>
    );
  }

  return null;
}
