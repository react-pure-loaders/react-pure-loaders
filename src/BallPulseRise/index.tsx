/** @jsx jsx */
import { jsx } from '@emotion/react';
import style from './style';
import { BaseProps } from '../types';

export default function BallPulseRise({ color, loading }: BaseProps) {
  if (loading) {
    return (
            <div css={style(color)}>
                <div/>
                <div/>
                <div/>
                <div/>
                <div/>
            </div>
    );
  }

  return null;
}
