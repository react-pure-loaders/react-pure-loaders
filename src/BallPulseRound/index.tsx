/** @jsx jsx */
import { jsx } from '@emotion/react';
import style from './style';
import { BaseProps } from '../types';

export default function BallPulseRound({ color, loading }: BaseProps) {
  if (loading) {
    return (
            <div css={style(color)}>
                <div/>
            </div>
    );
  }

  return null;
}
