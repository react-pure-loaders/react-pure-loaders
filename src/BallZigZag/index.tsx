/** @jsx jsx */
import { jsx } from '@emotion/react';
import style from './style';
import { BaseProps } from '../types';

export default function BallZigZag({ color, loading }: BaseProps) {
  if (loading) {
    return (
            <div css={style(color)}>
                <div/>
                <div/>
            </div>
    );
  }

  return null;
}
