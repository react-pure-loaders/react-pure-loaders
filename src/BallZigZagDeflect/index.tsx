/** @jsx jsx */
import { jsx } from '@emotion/react';
import style from './style';
import { BaseProps } from '../types';

export default function BallZigZagDeflect({ color, loading }: BaseProps) {
  if (loading) {
    return (
            <div css={style(color)}>
                <div/>
                <div/>
            </div>
    );
  }

  return null;
}
