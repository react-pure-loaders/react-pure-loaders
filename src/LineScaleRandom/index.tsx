/** @jsx jsx */
import { jsx } from '@emotion/react';
import style from './style';
import { BaseProps } from '../types';

// eslint-disable-next-line @typescript-eslint/explicit-function-return-type
export default function LineScaleRandom({ color, loading }: BaseProps) {
  if (loading) {
    return (
            <div css={style(color)}>
                <div />
                <div />
                <div />
                <div />
            </div>
    );
  }

  return null;
}
