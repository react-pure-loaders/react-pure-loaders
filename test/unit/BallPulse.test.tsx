import React from 'react';
import Chance from 'chance';
import { matchers } from '@emotion/jest';
import { render, cleanup } from '@testing-library/react';


import BallPulse from '../../src/BallPulse';
import { PRIMARY_COLOR } from '../../src/variables';

expect.extend(matchers);

const chance = new Chance();

describe('<BallPulse>', () => {
    afterEach(cleanup);

    test('BallPulse should match snapshot', () => {
        const { container } = render(<BallPulse loading={true}/>);

        expect(container.firstChild).toMatchSnapshot();
    });

    test('BallPulse should have default color', () => {
        const { container } = render(<BallPulse loading={true}/>);

        expect(container.firstChild).toHaveStyleRule('background-color', PRIMARY_COLOR, { target: '> div' });
    });

    test('BallPulse should have given color', () => {
        const color = chance.color({ format: 'hex' });
        const { container } = render(<BallPulse color={color} loading={true}/>);

        expect(container.firstChild).toHaveStyleRule('background-color', color, { target: '> div' });
    });

    test('BallPulse should have no children', () => {
        const { container } = render(<BallPulse loading={false}/>);

        expect(container.firstChild).toBeNull();
    });

    test('BallPulse should have three children', () => {
        const { container } = render(<BallPulse loading={true}/>);

        expect(container.querySelectorAll('div')).toHaveLength(4);
    });
});
