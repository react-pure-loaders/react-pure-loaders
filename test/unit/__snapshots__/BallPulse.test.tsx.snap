// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BallPulse> BallPulse should match snapshot 1`] = `
@keyframes animation-0 {
  0% {
    opacity: 1;
    -webkit-transform: scale(1);
    -moz-transform: scale(1);
    -ms-transform: scale(1);
    transform: scale(1);
  }

  45% {
    opacity: 0.7;
    -webkit-transform: scale(0.1);
    -moz-transform: scale(0.1);
    -ms-transform: scale(0.1);
    transform: scale(0.1);
  }

  80% {
    opacity: 1;
    -webkit-transform: scale(1);
    -moz-transform: scale(1);
    -ms-transform: scale(1);
    transform: scale(1);
  }
}

@keyframes animation-0 {
  0% {
    opacity: 1;
    -webkit-transform: scale(1);
    -moz-transform: scale(1);
    -ms-transform: scale(1);
    transform: scale(1);
  }

  45% {
    opacity: 0.7;
    -webkit-transform: scale(0.1);
    -moz-transform: scale(0.1);
    -ms-transform: scale(0.1);
    transform: scale(0.1);
  }

  80% {
    opacity: 1;
    -webkit-transform: scale(1);
    -moz-transform: scale(1);
    -ms-transform: scale(1);
    transform: scale(1);
  }
}

@keyframes animation-0 {
  0% {
    opacity: 1;
    -webkit-transform: scale(1);
    -moz-transform: scale(1);
    -ms-transform: scale(1);
    transform: scale(1);
  }

  45% {
    opacity: 0.7;
    -webkit-transform: scale(0.1);
    -moz-transform: scale(0.1);
    -ms-transform: scale(0.1);
    transform: scale(0.1);
  }

  80% {
    opacity: 1;
    -webkit-transform: scale(1);
    -moz-transform: scale(1);
    -ms-transform: scale(1);
    transform: scale(1);
  }
}

.emotion-0>div:nth-of-type(1) {
  -webkit-animation: animation-0 0.75s -0.24s infinite cubic-bezier(.2,.68,.18,1.08);
  animation: animation-0 0.75s -0.24s infinite cubic-bezier(.2,.68,.18,1.08);
}

.emotion-0>div:nth-of-type(2) {
  -webkit-animation: animation-0 0.75s -0.12s infinite cubic-bezier(.2,.68,.18,1.08);
  animation: animation-0 0.75s -0.12s infinite cubic-bezier(.2,.68,.18,1.08);
}

.emotion-0>div:nth-of-type(3) {
  -webkit-animation: animation-0 0.75s 0.00s infinite cubic-bezier(.2,.68,.18,1.08);
  animation: animation-0 0.75s 0.00s infinite cubic-bezier(.2,.68,.18,1.08);
}

.emotion-0>div {
  border-radius: 100%;
  height: 15px;
  margin: 2px;
  width: 15px;
  background-color: #ffffff;
  -webkit-animation-fill-mode: both;
  animation-fill-mode: both;
  display: inline-block;
}

<div
  class="emotion-0"
>
  <div />
  <div />
  <div />
</div>
`;
